wagtail/wagtailadmin/edit_handlers.py

```diff
@@ -25,26 +25,43 @@
 from wagtail.utils.deprecation import RemovedInWagtail11Warning
 
 
+# Form field properties to override whenever we encounter a model field
+# that matches one of these types - including subclasses
 FORM_FIELD_OVERRIDES = {
     models.DateField: {'widget': widgets.AdminDateInput},
     models.TimeField: {'widget': widgets.AdminTimeInput},
     models.DateTimeField: {'widget': widgets.AdminDateTimeInput},
-    TaggableManager: {'widget': widgets.AdminTagWidget}
+    TaggableManager: {'widget': widgets.AdminTagWidget},
+}
+
+# Form field properties to override whenever we encounter a model field
+# that matches one of these types exactly, ignoring subclasses.
+# (This allows us to override the widget for models.TextField, but leave
+# the RichTextField widget alone)
+DIRECT_FORM_FIELD_OVERRIDES = {
```

Comment:
Not sure how `wagtail_formfield` would help... firstly that would involve monkey-patching a core Django class, which is exactly what we (and the django.contrib.admin code we borrowed it from) are trying to avoid with the external overrides dict. Secondly, we'd still run into the question of whether we want subclasses to inherit the property. In the case of `RichTextField` we want `RichTextField.formfield` to take precedence over `TextField.wagtail_formfield`.

Creating subclasses of model fields [isn't something that people will be doing casually](https://docs.djangoproject.com/en/1.7/howto/custom-model-fields/), and when they do, there's about a fifty-fifty chance that they'll want to provide a dedicated form field that takes precedence over the one that Wagtail admin would select based on its own override rules. For that reason, I think forcing AdminAutoHeightTextInput on them (and making them jump through an extra hoop to over-override that) is not a good default, and we're better off having logic in wagtailadmin to effectively say 'only basic TextFields receive AdminAutoHeightTextInput'.